core/codegen/src/attribute/param/mod.rs

```diff
@@ -11,7 +11,7 @@ pub enum Parameter {
     Static(Name),
     Ignored(Dynamic),
     Dynamic(Dynamic),
-    Guard(Guard),
+    Guard(Box<Guard>),
```

Comment:
I don't think we should do this.

Reply:
I don't understand why it should not be boxed, but I suggest replacing it by :
```
#[allow(clippy::large_enum_variant)] // Gard should not be boxed even if it's 344 bytes longer than
                                     // Dynamic.
```
I don't like that comment, feel free to suggest me a better one : I try to explain why the lint shall be disabled because it should be the exception rather than the rule.

Reply:
It's simply a choice: do you pay the cost for boxing the value in exchange for cheaper copies of the box later, or do you eschew the cost of boxing but have more
expensive copies of the value later? In this case, we don't copy the structure around much, so I suspect we won't see a difference either way, and adding the Box here removes symmetry, so I consider it the worse option.